Formula/imagemagick.rb

```diff
@@ -46,11 +45,14 @@ class Imagemagick < Formula
   end
 
   on_linux do
+    depends_on "gcc"
     depends_on "libx11"
   end
 
   skip_clean :la
 
+  fails_with gcc: "5" # ghostscript is built with GCC
```

Comment:
We have been hitting another `pdfsandwich` failure after fixing issue with `libomp` (#103272). From debugging, it looks to be GCC related. Currently building local copy from source to see if error goes away.
```
     69062:	checking for version `GLIBCXX_3.4.29' in file /usr/lib/x86_64-linux-gnu/libstdc++.so.6 [0] required by file /home/linuxbrew/.linuxbrew/opt/ghostscript/lib/libgs.so.9 [0]
     69062:	/usr/lib/x86_64-linux-gnu/libstdc++.so.6: error: version lookup error: version `GLIBCXX_3.4.29' not found (required by /home/linuxbrew/.linuxbrew/opt/ghostscript/lib/libgs.so.9) (fatal)
     69062:
     69062:	file=/home/linuxbrew/.linuxbrew/Cellar/imagemagick/7.1.0-37_1/lib/ImageMagick/modules-Q16HDRI/coders/pdf.so [0];  destroying link map
```

Reply:
Local build fixed `pdfsandwich`. Hopefully this is enough to finally fix this issue.

May be able to avoid in future once we update base `gcc` version and reduce mix of system `gcc-5` and `gcc`, which can cause runtime issues if not properly detected in test.